File: android/src/main/java/ru/innim/flutter_login_vk/VKClient.java
package ru.innim.flutter_login_vk;

import com.vk.api.sdk.VK;
import com.vk.api.sdk.VKKeyValueStorage;
import com.vk.api.sdk.auth.VKAccessToken;
import com.vk.api.sdk.auth.VKScope;
import com.vk.sdk.api.users.dto.UsersFields;

import java.util.Arrays;
import java.util.List;
import java.util.Map;
import java.util.HashMap;

public class VKClient {

    static final List<UsersFields> FIELDS_DEFAULT = Arrays.asList(
            UsersFields.ONLINE,
            UsersFields.PHOTO_50,
            UsersFields.PHOTO_100,
            UsersFields.PHOTO_200
    );

    static final Map<VKScope, Integer> SCOPE_BYTES =
            new HashMap<VKScope, Integer>() {{
                put(VKScope.NOTIFY, 1);
                put(VKScope.FRIENDS, 2);
                put(VKScope.PHOTOS, 4);
                put(VKScope.AUDIO, 8);
                put(VKScope.VIDEO, 16);
                put(VKScope.STORIES, 64);
                put(VKScope.PAGES, 128);
                put(VKScope.STATUS, 1024);
                put(VKScope.NOTES, 2048);
                put(VKScope.MESSAGES, 4096);
                put(VKScope.WALL, 8192);
                put(VKScope.ADS, 32768);
                put(VKScope.OFFLINE, 65536);
                put(VKScope.DOCS, 131072);
                put(VKScope.GROUPS, 262144);
                put(VKScope.NOTIFICATIONS, 524288);
                put(VKScope.STATS, 1048576);
                put(VKScope.EMAIL, 4194304);
                put(VKScope.MARKET, 134217728);
            }};

    static boolean hasScopes(List<VKScope> scopes, int permission) {
        if (scopes != null && scopes.size() != 0) {
            int scopesMask = 0;
            for (VKScope scope : scopes) {
                final int value = SCOPE_BYTES.get(scope);
                scopesMask |= value;
            }

            return (scopesMask & permission) == scopesMask;
        }
        return true;
    }

    static VKAccessToken getCurrentAccessToken() {
        VKKeyValueStorage storage = VK.apiManager.getConfig().getKeyValueStorage();
        return VKAccessToken.Companion.restore(storage);
    }
}

File: android/src/main/kotlin/ru/innim/flutter_login_vk/VKClient.kt
package ru.innim.flutter_login_vk

import com.vk.api.sdk.auth.VKScope
import com.vk.sdk.api.users.dto.UsersFields


class VKClient {
    companion object {
        val fieldsDefault = listOf(
                UsersFields.ONLINE,
                UsersFields.PHOTO_50,
                UsersFields.PHOTO_100,
                UsersFields.PHOTO_200
        )

        private val scopeBytes = hashMapOf(
                VKScope.NOTIFY to 1,
                VKScope.FRIENDS to 2,
                VKScope.PHOTOS to 4,
                VKScope.AUDIO to 8,
                VKScope.VIDEO to 16,
                VKScope.STORIES to 64,
                VKScope.PAGES to 128,
                VKScope.STATUS to 1024,
                VKScope.NOTES to 2048,
                VKScope.MESSAGES to 4096,
                VKScope.WALL to 8192,
                VKScope.ADS to 32768,
                VKScope.OFFLINE to 65536,
                VKScope.DOCS to 131072,
                VKScope.GROUPS to 262144,
                VKScope.NOTIFICATIONS to 524288,
                VKScope.STATS to 1048576,
                VKScope.EMAIL to 4194304,
                VKScope.MARKET to 134217728
        )

        fun hasScopes(scopes: List<VKScope>, permission: Int): Boolean {
            if (scopes.isNotEmpty()) {
                var scopesMask = 0
                for (scope in scopes) {
                    val value: Int? = scopeBytes.get(key = scope)

                    if (value != null) {
                        scopesMask = scopesMask or value
                    }
                }
                return scopesMask and permission == scopesMask
            }
            return true
        }
    }
}